### PR TITLE
Issue 4388 - Don't draw pulse Schedule name by default when plotting

### LINKED ISSUE #4388: Don't draw pulse Schedule name by default when plotting
### What is the expected enhancement?
Pulse schedule names are currently being drawn by default. Most names are useless as they are automatically generated. Names should be drawn as an optional flag.

### PR DESCRIPTION
**Pull request by SeifMostafa**
…autogenerated title

### Summary

Fixes #4388

### Details and comments
Pulse schedule names are automatically generated, useless & drawn by default.
Added draw_title boolean with default value False
so only if draw_title is passed with value true, plot will have title.

- Parent class qiskit-terra/qiskit/pulse/pulse_lib/**Pulse**
- Schedule,ParametricPulse, SamplePulse are changed to be ready for new parameter

- **visualization.pulse_drawer** is used to redirect matplotlib drawer SamplePulse & Schedule so it changed to accept new parameter.
- at the end **matplotlib** class that contains ScheduleDrawer, SamplePulseDrawer modified to set plot title empty if draw_title is false.

Comments are updated each file contain method call.
following commands are run before pushing:
make style
make lint
make test

### CODE REVIEW

File: qiskit/pulse/pulse_lib/parametric_pulses.py [hunk 1]
```diff
@@ -103,7 +104,7 @@ def draw(self, dt: float = 1,
             interactive: When set true show the circuit in a new window
                 (this depends on the matplotlib backend being used supporting this)
             scaling: Deprecated, see `scale`
-
+            draw_title: When set 'True' plot will have title.
```
Comment (taalexander) [inline qiskit/pulse/pulse_lib/parametric_pulses.py:107]:
```suggestion
            draw_title: Add a title to the plot when set ``True``.
```

File: qiskit/pulse/pulse_lib/pulse.py [hunk 2]
```diff
@@ -68,6 +69,7 @@ def draw(self, dt: float = 1,
             interactive: When set true show the circuit in a new window
                 (this depends on the matplotlib backend being used supporting this)
             scaling: Deprecated, see `scale`
+            draw_title(Optional[bool]): When set 'True' plot will have title.
```
Comment (taalexander) [inline qiskit/pulse/pulse_lib/pulse.py:72]: There is no need to include the type hint here as it is in the signature.

Comment (taalexander) [inline qiskit/pulse/pulse_lib/pulse.py:72]:
```suggestion
            draw_title: Add a title to the plot when ``True``.
```

Comment (taalexander) [inline qiskit/pulse/pulse_lib/pulse.py:72]:
```suggestion
            draw_title: Add a title to the plot when set ``True``.
```

File: qiskit/pulse/pulse_lib/sample_pulse.py [hunk 3]
```diff
@@ -114,6 +115,7 @@ def draw(self, dt: float = 1,
             interactive: When set true show the circuit in a new window.
                          (This depends on the matplotlib backend being used.)
             scaling: Deprecated, see `scale`,
+            draw_title: When set 'True' plot will have title.
```
Comment (taalexander) [inline qiskit/pulse/pulse_lib/sample_pulse.py:118]:
```suggestion
            draw_title: Add a title to the plot when set ``True``.
```

File: qiskit/pulse/schedule.py [hunk 4]
```diff
@@ -447,7 +447,7 @@ def draw(self, dt: float = 1, style=None,
              interactive: bool = False, table: bool = True, label: bool = False,
              framechange: bool = True, scaling: float = None,
              channels: Optional[List[Channel]] = None,
-             show_framechange_channels: bool = True):
+             show_framechange_channels: bool = True, draw_title: bool = False):
```
Comment (taalexander) [inline qiskit/pulse/schedule.py:450]:
```suggestion
             show_framechange_channels: bool = True, 
             draw_title: bool = False):
```

File: qiskit/pulse/schedule.py [hunk 5]
```diff
@@ -468,6 +468,7 @@ def draw(self, dt: float = 1, style=None,
             scaling: Deprecated, see ``scale``.
             channels: A list of channel names to plot.
             show_framechange_channels: Plot channels with only framechanges.
+            draw_title: When set 'True' plot will have title.
```
Comment (taalexander) [inline qiskit/pulse/schedule.py:471]:
```suggestion
            draw_title: Add a title to the plot when set ``True``.
```

File: qiskit/visualization/pulse/matplotlib.py [hunk 6]
```diff
@@ -308,7 +310,10 @@ def draw(self, pulse: SamplePulse,
         bbox = ax.get_position()
 
         if self.style.title_font_size > 0:
-            figure.suptitle(str(pulse.name),
+            title = str('')
+            if draw_title:
+                title = str(pulse.name)
+            figure.suptitle(title,
```
Comment (taalexander) [inline qiskit/visualization/pulse/matplotlib.py:316]: Should this be indented in the conditional block?

File: qiskit/visualization/pulse/matplotlib.py [hunk 7]
```diff
@@ -863,9 +869,11 @@ def draw(self, schedule: ScheduleComponent,
             bbox = ax.get_position()
 
         if self.style.title_font_size > 0:
-            figure.suptitle(str(schedule.name),
+            title = str('')
+            if draw_title:
+                title = str(schedule.name)
+            figure.suptitle(title,
```
Comment (taalexander) [inline qiskit/visualization/pulse/matplotlib.py:875]: Same here?

File: qiskit/visualization/pulse_visualization.py [hunk 8]
```diff
@@ -80,10 +80,10 @@ def pulse_drawer(data: Union[SamplePulse, ScheduleComponent],
             All non-empty channels are shown if not provided.
         show_framechange_channels: When set `True` plot channels
             with only framechange instructions.
+        draw_title: When set 'True' plot will have title.
```
Comment (taalexander) [inline qiskit/visualization/pulse_visualization.py:83]:
```suggestion
        draw_title: Add a title to the plot when set ``True``.
```

File: qiskit/visualization/pulse/matplotlib.py (hunk 6 above)
Comment (SeifMostafa) [reply qiskit/visualization/pulse/matplotlib.py:316]: Yes,cause no where to call figure.suptitle out of the conditional block

File: qiskit/visualization/pulse/matplotlib.py (hunk 7 above)
Comment (SeifMostafa) [reply qiskit/visualization/pulse/matplotlib.py:875]: Yes, intented. Thanks!

**Comment by nkanazawa1989** [comment]
Looks good! One comment for the option name `draw_title`. We can change this to `title` because other parameter names are `label`, `table`, etc... and no explicit _draw_ before the option.

**Comment by SeifMostafa** [comment]
updated draw_title to title

**Comment by taalexander** [comment]
> Looks good! One comment for the option name `draw_title`. We can change this to `title` because other parameter names are `label`, `table`, etc... and no explicit _draw_ before the option.

I wonder if this is the right choice? The other flags do not conceivably need to be custom set. The `title` is currently implicitly the name of the Schedule. I could see the user wanting to override this. On the other hand, the user could always add a title to the plot manually if they disable it from being added. I'm fine with either decision @nkanazawa1989

**Comment by nkanazawa1989** [comment]
@taalexander That's a good point. If we want to add `draw_` to this, we need to update rest of option names. So reverting e50009e and making an another issue to update other option names?

**Comment by taalexander** [comment]
I'm not sure if we need to change the other names as it is less likely we need to change these?

**Comment by SeifMostafa** [comment]
You guys tell me which direction should I go & I will follow, Also you can mention me other issues to solve if you wanted

**Comment by nkanazawa1989** [comment]
@SeifMostafa Thanks. Let's use `draw_title` for now and see the feedback of users.

**Comment by SeifMostafa** [comment]
Resolved. I forgot calling with parameter in sample_pulse & parametric_pulses but was fixed in previous commit so I don't need reverting but adding new commit to revert title --> draw_title. 
All should be good now.

**Comment by nkanazawa1989** [comment]
Thank you! I think it's ready to go.

**Comment by 1ucian0** [comment]
Thanks @SeifMostafa for the fix! Do you mind having a look to the merging conflicts? Merged as soon as the are solved!

**Comment by SeifMostafa** [comment]
resolved

**Comment by taalexander** [comment]
Can we get the failing tests updated and then we can go ahead with a merge?

**Comment by SeifMostafa** [comment]
any pending on me?
tried to pull and do tests but it seems other changes committed with other imports and test result failed to reach 10/10 without searching and import those to solve errors.

**Comment by taalexander** [comment]
Seems like imports have changed and are causing conflicts. I am ok to merge upon passing tests.

**Comment by 1ucian0** [comment]
ping @SeifMostafa

**Comment by 1ucian0** [comment]
I'm closing this PR as stalled. Please, @SeifMostafa reopen if you are back! (check #4388 first, just in case)

**Comment by SeifMostafa** [comment]
Hello,
sorry for being late. Any hints please to remember what's required please
Would you like to set enhancement4388 changes on master and retest & submit new PR?
Thanks

**Comment by 1ucian0** [comment]
Hi! Welcome back!

> sorry for being late.

No worries.

> Any hints please to remember what's required please

I'm reopening. You can rebase and I think that would make it. Right @taalexander ?

**Comment by SeifMostafa** [comment]
rebase is challenged. 14 commits due changing parameter name many times. too many outdated codes. In my opinion, new, clear PR with clean, new branch to add draw_title boolean flag is good? Do you recommend that?

**Comment by 1ucian0** [comment]
Sure! as you prefer.

**Pull request closed without merging**